File: internal/database/model/stat.go
package model

import "github.com/topvennie/fragtape/pkg/sqlc"

type Result string

const (
	ResultWin     Result = "win"
	ResultLoss    Result = "loss"
	ResultTie     Result = "tie"
	ResultUnknown Result = ""
)

type Team string

const (
	TeamCT      Team = "ct"
	TeamT       Team = "t"
	TeamUnknown Team = ""
)

type Stat struct {
	ID        int
	DemoID    int
	UserID    int
	Result    Result
	StartTeam Team
	Kills     int
	Assists   int
	Deaths    int
}

func StatModel(s sqlc.Stat) *Stat {
	result := ResultUnknown
	if s.Result.Valid {
		result = Result(s.Result.Result)
	}
	startTeam := TeamUnknown
	if s.StartTeam.Valid {
		startTeam = Team(s.StartTeam.Team)
	}

	return &Stat{
		ID:        int(s.ID),
		DemoID:    int(s.DemoID),
		UserID:    int(s.UserID),
		Result:    result,
		StartTeam: startTeam,
		Kills:     fromInt(s.Kills),
		Assists:   fromInt(s.Assists),
		Deaths:    fromInt(s.Deaths),
	}
}

File: ui/src/lib/types/stat.ts
import { API } from "./api";

export enum Result {
  Win = "win",
  Loss = "loss",
  Tie = "tie",
}
export const resultString: Record<Result, string> = {
  [Result.Win]: "Win",
  [Result.Loss]: "Loss",
  [Result.Tie]: "Tie",
}

export enum Team {
  CT = "ct",
  T = "t",
}

export interface Stat {
  result: Result;
  startTeam: Team;
  kills: number;
  assists: number;
  deaths: number;
}

export const convertStat = (s: API.Stat): Stat => {
  return {
    result: s.result as Result,
    startTeam: s.start_team as Team,
    kills: s.kills,
    assists: s.assists,
    deaths: s.deaths,
  }
}
